File: Python/Balanced Binary Tree/Balanced Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if(not root):
            return True
        depth = self.helper(root)
        if(depth == -1):
            return False
        return True

    def helper(self, node):
        if(not node):
            return 0
        left = self.helper(node.left)
        right = self.helper(node.right)
        if(left == -1 or right == -1 or abs(left-right) > 1):
            return -1
        return 1+max(left, right)

File: C#/Balanced Binary Tree/Balanced Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    private bool isBalanced = true;

    public bool IsBalanced (TreeNode root) {
        if (root == null || (root.left == null && root.right == null)) return true;
        GetDepth (root);
        return isBalanced;
    }

    private int GetDepth (TreeNode node) {
        if (node == null) return 0;
        int leftDepth = GetDepth (node.left);
        int rightDepth = GetDepth (node.right);
        if (Math.Abs (leftDepth - rightDepth) > 1)
            isBalanced = false;
        return 1 + Math.Max (leftDepth, rightDepth);
    }
}